internal/venice-common/src/main/java/com/linkedin/venice/pubsub/api/PubsubProduceResult.java

```diff
@@ -0,0 +1,33 @@
+package com.linkedin.venice.pubsub.api;
+
+/**
+ * An interface implemented by specific PubsubProducerAdapters to return the result of a produce action.
+ */
+public interface PubsubProduceResult {
+  /**
+   * The offset of the record in the topic/partition.
+   */
+  long offset();
```

Comment:
I think we'll need to make this a generic type, even though for now the rest of the code assumes this is always a long.